spec/support/idv_examples/cancel_at_idv_step.rb

```diff
@@ -20,6 +20,15 @@
   end
 
   it 'shows the user a cancellation message with the option to cancel and reset idv' do
+    failure_to_proof_url = 'https://www.example.com/failure'
+    sp_name = 'Test SP'
+    allow_any_instance_of(SessionDecorator).to receive(:failure_to_proof_url).
```

Comment:
`SessionDecorator` should not be called in this scenario because there is an SP in the session. Only `ServiceProviderSessionDecorator` should be called.

Reply:
Ah, right. This shared example is used in 3 contexts: by visiting the site directly, via an OIDC SP, and via a SAML SP. So, in the SP cases, the name and URLs will be different for OIDC vs SAML, and in the case where there is no SP, it should maybe default to going to the account page like it used to.

By mocking the decorators, we are masking a bug in the scenario where there is no SP and we end up with a link called `‹ Back to` that points to `/verify/cancel`.

I would use a view spec with those 3 different scenarios.

Reply:
Or maybe just 2 scenarios: with and without SP.

Reply:
How about a simple conditional checking for sp_name

Reply:
and sending to accounts page as before

Reply:
yes.  sounds good.